src/index.js

```diff
@@ -73,7 +73,16 @@ function select(algorithm) {
 function deriveKeypair(seed, options) {
   const decoded = addressCodec.decodeSeed(seed)
   const algorithm = decoded.type === 'ed25519' ? 'ed25519' : 'ecdsa-secp256k1'
-  return select(algorithm).deriveKeypair(decoded.bytes, options)
+  const method = select(algorithm)
+  const keypair = method.deriveKeypair(decoded.bytes, options)
+  const goodMessage = hash('This test message should verify.')
+  const badMessage = hash('This test message should NOT verify.')
+  const goodSignature = method.sign(goodMessage, keypair.privateKey)
+  if (method.verify(goodMessage, goodSignature, keypair.publicKey) !== true ||
+      method.verify(badMessage, goodSignature, keypair.publicKey) !== false) {
```

Comment:
I'm not sure how meaningful the second check is. It seems like it tells us more about the correctness of the signing and verification functionality than the generated keypair.

Reply:
True. I'll go ahead and remove it.